tests/Metadata/Driver/YamlDriverTest.php

```diff
@@ -72,12 +74,45 @@ public function testBlogPostAccessor()
         self::assertEquals($p, $m->propertyMetadata['title']);
     }
 
-    private function getDriverForSubDir($subDir = null)
+    /**
+     * @expectedException  \JMS\Serializer\Exception\InvalidMetadataException
+     */
+    public function testInvalidMetadataFileCausesException(): void
     {
-        return new YamlDriver(new FileLocator([
+        $this->getDriverForSubDir('invalid_metadata')->loadMetadataForClass(new \ReflectionClass(BlogPost::class));
+    }
+
+    public function testLoadingYamlFileWithLongExtension(): void
+    {
+        $m = $this->getDriverForSubDir('multiple_types')->loadMetadataForClass(new \ReflectionClass(Person::class));
+
+        self::assertArrayHasKey('name', $m->propertyMetadata);
+    }
+
+    public function testLoadingMultipleMetadataExtensions(): void
+    {
+        $classNames = $this->getDriverForSubDir('multiple_types', false)->getAllClassNames();
+
+        self::assertEquals(
+            [
+                BlogPost::class,
+                Person::class
+            ],
+            $classNames
+        );
+    }
+
+    private function getDriverForSubDir($subDir = null, bool $addUnderscoreDir = true): YamlDriver
+    {
+        $dirs = [
             'JMS\Serializer\Tests\Fixtures' => __DIR__ . '/yml' . ($subDir ? '/' . $subDir : ''),
-            '' => __DIR__ . '/yml/_' . ($subDir ? '/' . $subDir : ''),
-        ]), new IdenticalPropertyNamingStrategy(), null, $this->getExpressionEvaluator());
+        ];
+
+        if ($addUnderscoreDir) {
```

Comment:
This one broke `getAllClassNames` testm since this is invalid dir for most cases, but only `getAllClassNames` iterates over all directories and fails. So I made this optional